bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/DeviceHandler.java

```diff
@@ -0,0 +1,345 @@
+import static org.openhab.binding.salus.internal.SalusBindingConstants.SalusDevice.DSN;
+import static org.openhab.core.thing.ThingStatus.OFFLINE;
+import static org.openhab.core.thing.ThingStatus.ONLINE;
+import static org.openhab.core.thing.ThingStatusDetail.BRIDGE_UNINITIALIZED;
+import static org.openhab.core.thing.ThingStatusDetail.COMMUNICATION_ERROR;
+import static org.openhab.core.thing.ThingStatusDetail.CONFIGURATION_ERROR;
+import static org.openhab.core.types.RefreshType.REFRESH;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedSet;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.salus.internal.SalusBindingConstants;
+import org.openhab.binding.salus.internal.rest.DeviceProperty;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.OpenClosedType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.types.UpDownType;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Martin Grześlowski - Initial contribution
+ */
+@NonNullByDefault
+public class DeviceHandler extends BaseThingHandler {
+    private static final BigDecimal ONE_HUNDRED = new BigDecimal(100);
+    private final Logger logger;
+    @NonNullByDefault({})
+    private String dsn;
+    @NonNullByDefault({})
+    private CloudApi cloudApi;
+    private final Map<String, String> channelUidMap = new HashMap<>();
+    private final Map<String, String> channelX100UidMap = new HashMap<>();
+
+    public DeviceHandler(Thing thing) {
+        super(thing);
+        logger = LoggerFactory.getLogger(DeviceHandler.class.getName() + "[" + thing.getUID().getId() + "]");
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            internalInitialize();
+        } catch (Exception e) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR,
+                    "Error occurred while initializing Salus device! " + e.getLocalizedMessage());
+        }
+    }
+
+    private void internalInitialize() {
+        var bridge = getBridge();
+        if (bridge == null) {
+            updateStatus(OFFLINE, BRIDGE_UNINITIALIZED,
+                    "There is no bridge for this thing. Remove it and add it again.");
```

Comment:
Can you make this string avilable to i18n? See earlier comment. This file has more `updateSTatus` calls, please also adapt those.

Reply:
Just so I know:

Is there a fast way to translate `"There is no " + DSN + " for this thing. Remove it and add it again."` with `@text/...` or I need to use `TranslationProvider` object?

Reply:
I think the `@text/... `  should be able to handle parameters, but i don't have an example ready to share.

Reply:
OK, I found this pair as an example:

`@text/offline.api2.comm-error.exception [\"" + e.getMessage() + "\"]`

`offline.api2.comm-error.exception = An unexpected exception occurred: {0}`

Reply:
Ah, thanks!